felix/calc/encapsulation_resolver.go

```diff
@@ -251,25 +251,33 @@ func (c *EncapsulationCalculator) removePool(cidr string) {
 }
 
 func (c *EncapsulationCalculator) IPIPEnabled() bool {
+	if len(c.ipipPools) > 0 {
+		return true
+	}
 	if c.config != nil && c.config.IpInIpEnabled != nil {
 		return *c.config.IpInIpEnabled
 	}
 
-	return len(c.ipipPools) > 0
+	return false
 }
 
 func (c *EncapsulationCalculator) VXLANEnabled() bool {
+	if len(c.vxlanPools) > 0 {
+		return true
+	}
 	if c.config != nil && c.config.VXLANEnabled != nil {
 		return *c.config.VXLANEnabled
 	}
-
-	return len(c.vxlanPools) > 0
+	return false
 }
 
 func (c *EncapsulationCalculator) VXLANEnabledV6() bool {
+	if len(c.vxlanPoolsv6) > 0 {
+		return true
+	}
 	if c.config != nil && c.config.VXLANEnabled != nil {
 		return *c.config.VXLANEnabled
 	}
```

Comment:
@muff1nman I'm thinking in order to fix https://github.com/projectcalico/calico/issues/6670 we could leave the rest of this file as it was (so revert all changes) + remove these lines. My intention would be to decouple the EncapsulationCalculator logic for IPv6 VXLAN from FelixConfig entirely, as it doesn't need an "override" like the IPv4 encapsulations.

We'd also need some test changes to conform to that.

Reply:
Okay sounds good, let me try going down that route